app/lib/timetable/src/edit_time.dart

```diff
@@ -114,8 +115,9 @@ Future<Time> selectTime(BuildContext context,
   return showIntervalTimePicker(
     context: context,
     initialTime: initialTime?.toTimeOfDay() ?? TimeOfDay(hour: 9, minute: 10),
-    interval: 30,
-    visibleStep: VisibleStep.thirtieths,
+    interval: minutesInterval,
+    visibleStep:
+        minutesInterval == 30 ? VisibleStep.thirtieths : VisibleStep.fifths,
```

Comment:
If we're basically hard coding the logic here that it's either 30 or 5 minute intervals then shouldn't we also also add an assert that `minutesInteval` is one or the other?

Reply:
I'm using an extension now on `int` to get the `VisibleStep` and throw an exception if it's an unknown minute value:

```dart
extension on int {
  VisibleStep toVisibleStep() {
    switch (this) {
      case 1:
      case 5:
        return VisibleStep.fifths;
      case 6:
        return VisibleStep.sixths;
      case 10:
        return VisibleStep.tenths;
      case 20:
        return VisibleStep.twentieths;
      case 30:
        return VisibleStep.thirtieths;
      case 60:
        return VisibleStep.sixtieth;
      default:
        // At the moment, only these intervals are supported. If you need
        // another one, please add it and handle it in the switch statement.
        throw Exception(
          "Unsupported minutes interval: $this. Please handle the other cases yourself.",
        );
    }
  }
}
```